pytradfri/api/libcoap_api.py

```diff
@@ -98,7 +110,9 @@ def _execute(self, api_command, *, timeout=None):
         api_command.process_result(_process_output(return_value, parse_json))
         return api_command.result
 
-    def request(self, api_commands, *, timeout=None):
+    def request(
+        self, api_commands: list[Command[T]], *, timeout: int | None = None
+    ) -> T | list[T] | list[Any] | None:
         """Make a request. Timeout is in seconds."""
         if not isinstance(api_commands, list):
```

Comment:
Do we need this? Don't we expect api_commands to always be a list given the type hint for `api_commands`?

Reply:
Check the typing of the async api. It should be the same, including overrides.

Reply:
I add to make some adjustments as `_execute` can also return `None`.

Reply:
No, execute should always return a command result. When starting an observation we should also return the command result.